backend/director/tools/muapi_video.py

```
@@ -0,0 +1,210 @@
+        )
+
+    def _submit(self, endpoint: str, payload: dict) -> str:
+        """Submit a generation request and return the request_id."""
+        resp = self.session.post(f"{self.BASE_URL}/{endpoint}", json=payload, timeout=30)
+        resp.raise_for_status()
+        try:
+            request_id = resp.json()["request_id"]
+        except KeyError:
+            raise Exception(
+                f"MuAPI did not return a request_id. Response: {resp.text}"
+            )
```

Ruff (0.15.18)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)